### PR TITLE
Integration of the Qt-GUI

### PR DESCRIPTION
**Pull request by alexandrospopov**
Integrate core function and classes of the qt viewer.

### CODE REVIEW

File: scripts/viewer2.py [hunk 1]
```diff
@@ -0,0 +1,880 @@
+import sys
+import PyQt4.QtGui as QtGui
+import PyQt4.QtCore as QtCore
+
+import sys
```
Comment (peristeri) [inline scripts/viewer2.py]: You importing sys twice.

File: scripts/viewer2.py [hunk 2]
```diff
@@ -0,0 +1,880 @@
...
+from msct_parser import Parser
+from msct_image import Image
+from bisect import bisect
+from numpy import arange, max, pad, linspace, mean, median, std, percentile
+import numpy as np
+from msct_types import *
```
Comment (peristeri) [inline scripts/viewer2.py]: Explicit import of msct_types functions.

File: scripts/viewer2.py [hunk 3]
```diff
@@ -0,0 +1,880 @@
...
+
+from matplotlib.widgets import Slider, Button, RadioButtons
+from matplotlib.backends.backend_qt4agg import FigureCanvasQTAgg as FigureCanvas
+
+from abc import ABCMeta, abstractmethod
+
+import webbrowser
+
+
+class Observer(object):
```
Comment (peristeri) [inline scripts/viewer2.py]: You mentioned that Observer and Observable are not used. Is this still the case?

File: scripts/viewer2.py [hunk 4]
```diff
@@ -0,0 +1,880 @@
...
+    def unregister_all(self):
+        if self.observers:
+            del self.observers[:]
+
+    def update_observers(self, *args, **kwargs):
+        for observer in self.observers:
+            observer.update_observer(*args, **kwargs)
+
+
+class SinglePlot():
```
Comment (peristeri) [inline scripts/viewer2.py]: `SinglePlot` is not a descriptive name. Maybe use something like `BaseAnatomicalWindow`

File: scripts/viewer2.py [hunk 5]
```diff
@@ -0,0 +1,880 @@
...
+            index_max = np.argmax(array_dim)
+            max_size = array_dim[index_max]
+            self.offset = [int(round((max_size - array_dim[0]) / self.im_spacing[0]) / 2),
+                           int(round((max_size - array_dim[1]) / self.im_spacing[1]) / 2),
+                           0]
+
+    def is_point_in_image(self, target_point):
+        return 0 <= target_point.x < self.image_dim[0] and 0 <= target_point.y < self.image_dim[1] and 0 <= target_point.z < self.image_dim[2]
+
+class SinglePlotMain(SinglePlot,Observer):
```
Comment (peristeri) [inline scripts/viewer2.py]: Please use a more descriptive name.

File: scripts/viewer2.py [hunk 6]
```diff
@@ -0,0 +1,880 @@
...
+            if ipoints.x == self.current_slice:
+                x,y=select_right_dimensions(ipoints,self.view)
+                x_data.append(x)
+                y_data.append(y)
+        self.plot_points.set_xdata(x_data)
+        self.plot_points.set_ydata(y_data)
+        self.refresh()
+
+class SinglePlotSecond(SinglePlot,Observer,object):
+    #TODO : inheritance etrange
```
Comment (peristeri) [inline scripts/viewer2.py]: Not sure what this todo is suppose to mean.

**Comment by jcohenadad** [comment]
hello, is it ready to be tested?

**Comment by peristeri** [comment]
@jcohenadad not yet.

File: scripts/sct_propseg.py.orig [hunk 7]
```diff
@@ -0,0 +1,598 @@
+#!/usr/bin/env python
```
Comment (peristeri) [inline scripts/sct_propseg.py.orig]: Delete this file

File: spinalcordtoolbox/gui.py [hunk 8]
```diff
@@ -0,0 +1,880 @@
+import sys
```
Comment (peristeri) [inline spinalcordtoolbox/gui.py]: Delete this file

File: tests/test_viewer.py [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import spinalcordtoolbox.gui as gui
```
Comment (peristeri) [inline tests/test_viewer.py]: Delete this file

**Comment by SaraDupont** [comment]
What is the status of this PR ? The new version of `sct_make_ground_truth` would be useful for Eunice, do we need to find a workaround or will this be merged to master in the near future ?

**Comment by peristeri** [comment]
We are in the process of finishing the refactoring of the viewer. I will do a presentation of where we stand in tomorrow's meeting.

**Pull request closed without merging**